src/libs/actions/Session/index.ts

```diff
@@ -637,7 +636,6 @@ function resetHomeRouteParams() {
 function cleanupSession() {
     Pusher.disconnect();
     Timers.clearAll();
-    Welcome.resetReadyCheck();
```

Comment:
No need for that. `resetReadyCheck` was used to reset state so that next time user logs in, he starts with a fresh state and the promise is reset and so on. 

With the new implementation in Welcome.tsx we always simply check the appropriate ONYXKEYS to determine whether to show modal or not, no need to reset anything before logout.

Here's the old implementation for reference:
```
function resetReadyCheck() {
    isReadyPromise = new Promise((resolve) => {
        resolveIsReadyPromise = resolve;
    });
    isFirstTimeNewExpensifyUser = undefined;
    isLoadingReportData = true;
}
```